iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/AbstractLimitsCheck.java

```diff
@@ -39,14 +40,19 @@ void registerObjectCheck() {
   void reportMissingLimit(BlockObject container) {
     container.block("resources").block("limits")
       .attribute(getLimitAttributeKey())
-      .reportIfAbsent(getFirstChildElement(container), getMessage())
+      .reportIfAbsent(retrieveTextRangeToRaiseIssue(container), getMessage())
       .reportIfValue(isSet().negate(), getMessage());
   }
 
   @Nullable
-  static HasTextRange getFirstChildElement(BlockObject blockObject) {
+  static HasTextRange retrieveTextRangeToRaiseIssue(BlockObject blockObject) {
     if (blockObject.tree != null) {
-      return blockObject.tree.elements().get(0).key().metadata();
+      return blockObject.tree.elements().stream()
+        .filter(element -> element.key() instanceof ScalarTreeImpl)
+        .filter(element -> "spec".equals(((ScalarTreeImpl) element.key()).value()))
```

Comment:
This implementation is highly specific to `spec` and does not fit with the theme of the `AbstractLimitCheck`. I would leave it as it is, and implement the required logic in the Check itself.